pallets/schemas/src/lib.rs

```diff
@@ -611,6 +651,280 @@ pub mod pallet {
+
+			let proposal: Box<T::Proposal> = Box::new(
+				(Call::<T>::create_intent_group_via_governance {
+					creator_key: proposer.clone(),
+					intent_group_name,
+					intent_ids,
+				})
+				.into(),
+			);
+			T::ProposalProvider::propose_with_simple_majority(proposer, proposal)?;
+			Ok(())
+		}
+
+		/// Update an IntentGroup (testnet)
+		///
+		/// # Events
+		/// * [`Event::IntentGroupUpdated`]
+		///
+		/// # Errors
+		/// * [`Error::InvalidIntentGroupId`] - The specified [`IntentGroupId`] does not exist.
+		/// * [`Error::InvalidIntentId`] - At least one of the specified [`IntentId`]s does not exist.
+		/// * [`Error::TooManyIntentsInGroup`] - The update would result in too many Intents in the group.
+		#[pallet::call_index(16)]
+		#[pallet::weight(T::WeightInfo::update_intent_group(intent_ids.len() as u32))]
+		pub fn update_intent_group(
+			origin: OriginFor<T>,
+			intent_group_id: IntentGroupId,
+			intent_ids: BoundedVec<IntentId, T::MaxIntentsPerIntentGroup>,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+
+			Self::update_intent_group_for(intent_group_id, intent_ids)?;
+			Self::deposit_event(Event::IntentGroupUpdated { key: sender, intent_group_id });
+			Ok(())
+		}
+
+		/// Update an IntentGroup by means of council approval
+		///
+		/// # Events
+		/// * [`Event::IntentGroupUpdated`]
+		///
+		/// # Errors
+		/// * [`Error::InvalidIntentGroupId`] - The specified [`IntentGroupId`] does not exist.
+		/// * [`Error::InvalidIntentId`] - At least one of the specified [`IntentId`]s does not exist.
+		/// * [`Error::TooManyIntentsInGroup`] - The update would result in too many Intents in the group.
+		///
+		#[pallet::call_index(17)]
+		#[pallet::weight(
+            T::WeightInfo::update_intent_group_via_governance(intent_ids.len() as u32)
+        )]
+		pub fn update_intent_group_via_governance(
+			origin: OriginFor<T>,
+			updater_key: T::AccountId,
+			intent_group_id: IntentGroupId,
+			intent_ids: BoundedVec<IntentId, T::MaxIntentsPerIntentGroup>,
+		) -> DispatchResult {
+			T::CreateSchemaViaGovernanceOrigin::ensure_origin(origin)?;
+			ensure!(
+				IntentGroups::<T>::contains_key(intent_group_id),
+				Error::<T>::InvalidIntentGroupId
+			);
+			Self::update_intent_group_for(intent_group_id, intent_ids)?;
+
+			Self::deposit_event(Event::IntentGroupUpdated { key: updater_key, intent_group_id });
+			Ok(())
+		}
+
+		/// Propose to update an Intent Group. Creates a proposal for council approval to update an existing Intent Group.
+		///
+		#[pallet::call_index(18)]
+		#[pallet::weight(T::WeightInfo::propose_to_update_intent_group())]
+		pub fn propose_to_update_intent_group(
+			origin: OriginFor<T>,
+			intent_group_name: crate::types::SchemaNamePayload,
+			intent_ids: BoundedVec<IntentId, T::MaxIntentsPerIntentGroup>,
+		) -> DispatchResult {
+			let proposer = ensure_signed(origin)?;
+
+			let proposal: Box<T::Proposal> = Box::new(
+				(Call::<T>::create_intent_group_via_governance {
```

Comment:
this ties to my previous question, how update operation is handled by `create_intent_group_via_governance` since it creates a new group id everytime ?